lib/reply.js

```diff
@@ -532,12 +522,13 @@ function handleError (reply, error, cb) {
     return
   }
 
-  const payload = serializeError({
+  // TODO shallow copy bad idea in hot path
+  const payload = serializeError(reply.context, {
     error: statusCodes[statusCode + ''],
-    code: error.code,
     message: error.message || '',
-    statusCode: statusCode
-  })
+    ...error,
+    statusCode
+  }, statusCode)
```

Comment:
Tehcnically this is not an `Error` anymore. A good "trick" to avoid copying could be to use `const child = Object.create(error)` and then set the properties on the child.